src/main/java/org/benetech/servicenet/matching/counter/LocationSimilarityCounter.java

```diff
@@ -33,68 +35,100 @@
     @Value("${similarity-ratio.weight.location.in-level-3-distance}")
     private float level3distanceRatio;
 
+    @Value("${similarity-ratio.weight.location.in-same-city-zipcode}")
+    private float sameCityOrZipCodeRatio;
+
     @Autowired
     private GeocodingResultService geocodingResultService;
 
     @Override
     public float countSimilarityRatio(Location location1, Location location2, MatchingContext context) {
-        if (anyPhysicalAddressIsNull(location1, location2)) {
+        if (anyPhysicalAddressIsNull(location1, location2) || areLocations255AddressTextBlank(location1, location2)) {
             return NO_MATCH_RATIO;
         }
 
-        if (allCoordinatesArePresent(location1, location2)) {
-            return countRatioIfCoordinatesArePresent(location1, location2);
-        } else {
-            return countRatioIfCoordinatesAreMissing(location1, location2, context);
+        List<GeocodingResult> baseGeocoding = getGeoCodingResult(location1, context);
+        List<GeocodingResult> partnerGeocoding = getGeoCodingResult(location2, context);
+
+        if (areCoordsNotComparable(baseGeocoding, partnerGeocoding, location1, location2)) {
```

Comment:
same here, it would be better to have "areCoordsComparable" method

Reply:
fixed